roles/mongodb-common/tasks/main.yml

```diff
@@ -7,6 +7,17 @@
   register: result
   until: result|succeeded
   retries: 5
+  when: os == 'ubuntu'
+
+- name: install mongodb packages on rhel
```

Comment:
how will this work on centos? I think you need to disable epel repo only when: ( os == 'rhel' and openstack_install_method == 'distro' )